devel/validate-platforms

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+#
+# Validate different platform builds of the final Nextstrain image.
+#
+set -euo pipefail
+
+# Set default values.
+registry=localhost:5000
```

Comment:
The [force-push](https://github.com/nextstrain/docker-base/compare/468cb918971f161fce0f9df54c34770614a2f9ee..c70aa6c2d37aed5572ae20f1fdfbd0c287559bb7) adds a snippet in the README to make this more apparent.

I manually tested with `-r docker.io` since that's the easiest way to showcase a reproducible success and failure. Going forwards, I'd expect usage to be primarily CI (where `-r ghcr.io` is used for many other scripts), and the occasional local development (where the localhost default would be consistently convenient).